.clang-tidy

```diff
@@ -0,0 +1 @@
+Checks: 'google-*'
```

Comment:
This is going to be too broad. Since the provenance of most of the code is going to be code from Chromium and Fuchsia, it never assumed to follow those stricter guidelines. Lets stick to the default non-google ones for now and opt-into each stricter check one by one while updating the codebase in the prcess.

Reply:
The default ones are just memory warnings which we violate in a few places.  I'd rather make this a team effort were we just clean things up as we go.  There really wasn't anything bad when I ran the full scan, missing brackets, c-style casts, non-const references for example.  If they ever get in the way of someone they can be turned off with comments or extra .clang-tidy files can be added.  What do you say?  Otherwise, I'm going to have to spend a day or 2 cleaning stuff up on my own.

Reply:
How would adding less restrictive rules make you spend more time? My thought was you'd commit this right away with the least restrictive rules and each one of us picks a rule category, adds it to this list and fixes the code to adhere to those rules.

But your call. If you want to land this as-is. Go for it.

Reply:
That would assume we have to address all violations before we turned on any check which isn't the case since it only runs on files that have changed.  That means new submissions aren't getting the benefit of the lint until someone slogs through migrating everything over.  The economics of that choice leaves us with less linting.  I'd rather get the benefit of the best linting for all new code and try to move over everything without forcing someone to do upfront migration.